chain/cosmos/cosmos_chain.go

```diff
@@ -780,11 +783,14 @@ func (c *CosmosChain) Timeouts(ctx context.Context, height uint64) ([]ibc.Packet
 
 // FindTxs implements blockdb.BlockSaver.
 func (c *CosmosChain) FindTxs(ctx context.Context, height uint64) ([]blockdb.Tx, error) {
+	c.findTxLock.Lock()
+	defer c.findTxLock.Unlock()
```

Comment:
So is this mutex supposed to guard against `FindTxs` being called before `StartAllNodes` or after `StopAllNodes`?

A mutex is probably simpler than using `sync.Atomic`, but this could be an appropriate use of an `RWMutex` so that `FindTxs` can be called concurrently.

Reply:
This detected race is with the `tn.Client` being set in `StartAllNodes` simultaneous with the `tn.Client` being used in `FindTxs`, but we also don't need the FindTxs calls to happen while the chain is stopped. But, isolating the mutex to the `StartAllNodes` might be enough to remove the race condition and shouldn't limit the flow to being `StopAllNodes` -> `StartAllNodes`

Reply:
kept mutex usage isolated to specific funcs